src/lib/optimizerTabController.js

```diff
@@ -94,11 +94,27 @@ export const OptimizerTabController = {
       if (event.data && fieldValues.characterId) {
         const character = DB.getCharacterById(fieldValues.characterId)
 
-        if (data.id) {
+        if (!data.id) {
+          window.optimizerGrid.current.api.deselectAll()
+        }
+
+        if (character && data.id) {
           const rowId = data.id
           const build = OptimizerTabController.calculateRelicsFromId(rowId)
           window.setOptimizerBuild(build)
-          selectRow(data.id)// feels kinda hacky but it works ¯\_(ツ)_/¯
+
+          // Find the row by its string ID and select it
+          const rowNode = window.optimizerGrid.current.api.getRowNode(String(data.id))
+          if (rowNode) {
+            const currentPinned = window.optimizerGrid.current.api.pinnedRowModel.pinnedTopRows.map((x) => x.data)
+
+            if (String(currentPinned[0].id) == String(rowNode.data.id)) {
```

Comment:
after equipping an optimizer build, the top row will correspond to an optimizer row